src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp

```diff
@@ -136,3 +143,201 @@ std::vector<Profile> WslDistroGenerator::GenerateProfiles()
+    DWORD dwNumSubKeys = 0;
+    if (RegQueryInfoKey(wslRootKey.get(),
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        &dwNumSubKeys,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr,
+                        nullptr) != ERROR_SUCCESS)
+    {
+        return false;
+    }
+
+    wchar_t buffer[39]; // a {GUID} is 38 chars long
+    for (DWORD i = 0; i < dwNumSubKeys; i++)
+    {
+        DWORD length;
+        const auto result = RegEnumKeyEx(wslRootKey.get(), i, &buffer[0], &length, nullptr, nullptr, nullptr, nullptr);
+        if (result == ERROR_NO_MORE_ITEMS)
+        {
+            break;
+        }
+
+        if (result == ERROR_SUCCESS &&
+            length == 38 &&
+            buffer[0] == L'{' &&
+            buffer[37] == L'}')
+        {
+            guidStrings.emplace_back(&buffer[0], length);
+        }
+    }
+
+    return true;
+}
+
+// Function Description:
+// - Get the list of all the names of all the WSL distros from the registry. If
+//   we fail to open any regkey for the GUID of a distro, we'll just skip it.
+//   Places the names of all the distros into the "names" param.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - guidStrings: A list of all the GUIDs of the installed distros
+// - names: a vector that receives all the names of the installed distros.
+// Return Value:
+// - false if the root key was invalid, else true.
+static bool getWslNames(const wil::unique_hkey& wslRootKey,
+                        std::vector<std::wstring>& guidStrings,
+                        std::vector<std::wstring>& names)
+{
+    if (!wslRootKey)
+    {
+        return false;
+    }
+    for (const auto& guid : guidStrings)
+    {
+        wil::unique_hkey distroKey{ openDistroKey(wslRootKey, guid) };
+        if (!distroKey)
+        {
+            continue;
+        }
+
+        std::wstring buffer;
+        auto result = wil::AdaptFixedSizeToAllocatedResult<std::wstring, 256>(buffer, [&](PWSTR value, size_t valueLength, size_t* valueLengthNeededWithNull) -> HRESULT {
+            auto length = static_cast<DWORD>(valueLength);
+            const auto status = RegQueryValueExW(distroKey.get(), RegKeyDistroName, 0, nullptr, reinterpret_cast<BYTE*>(value), &length);
+            // length will receive the number of bytes - convert to a number of
+            // wchar_t's. AdaptFixedSizeToAllocatedResult will resize buffer to
+            // valueLengthNeededWithNull
+            *valueLengthNeededWithNull = (length / sizeof(wchar_t));
+            // If you add one for another trailing null, then there'll actually
+            // be _two_ trailing nulls in the buffer.
+            return status == ERROR_MORE_DATA ? ERROR_SUCCESS : status;

```

Comment:
I'm not sure these are HRESULTs

Reply:
(I am also not certain I understand how this function works/what it is expecting)